src/vsr.zig

```diff
@@ -187,7 +187,18 @@ pub const Operation = enum(u8) {
 /// We reuse the same header for both so that prepare messages from the primary can simply be
 /// journalled as is by the backups without requiring any further modification.
 pub const Header = extern struct {
-    const checksum_body_empty = checksum(&.{});
+    var checksum_empty_once = std.once(checksum_empty_init);
+    var checksum_empty: u128 = undefined;
+    fn checksum_empty_init() void {
+        checksum_empty = checksum(&.{});
+    }
+
+    /// Aegis128 (used by checksum) uses hardware accelerated AES via inline asm which isn't
```

Comment:
Did a bit of digging in to this - Zig's AES will use hardware accelerated AES either via AESNI or arm64's extensions if supported by the target CPU.

However - in benchmark.sh, we don't specify a target CPU - so I believe it'll use whatever is available on the machine building (I could be wrong on this - at the very least it uses something more than `baseline`) - but everywhere else we explicitly specify `baseline` as our target CPU, which doesn't get AES hardware acceleration.

I think we should specify a more modern CPU target all round...

Reply:
After chatting a bit with Joran (this is mostly off topic for this PR haha):
- We should build our binaries for a relatively modern CPU target (we do expect people to have NVMe storage, after all)
- If you're on an older CPU, or something bleeding edge, you can compile yourself and it'll use `native` CPU. We should document this.

Coming back to this PR - I think we can bump our baseline to `x86_64_v3+aes` and implement the above, I'm happy to do that in a separate PR